index.html

```diff
@@ -5007,13 +4983,259 @@ <h3 id="looping-AudioBufferSourceNode">
+            subsequent sample frame in <code>buffer</code>, then the next
+            sample frame should be taken as the playback signal at
+            <code>actualLoopStart</code> for interpolation purposes.
+            </li>
+            <li>Let <code>bufferTime</code> equal -1. This is the <em>playhead
+            position</em> within <code>buffer</code> of the next output sample
+            frame. The initial value -1 indicates that this position has not
+            yet been determined.
+            </li>
+            <li>For each block of audio to be rendered:
+              <ol>
+                <li>Let <code>currentTime</code> be the value of
+                <code>currentTime</code> attribute of the node's
+                <a>AudioContext</a>.
+                </li>
+                <li>Let <code>contextRate</code> be the value of the
+                <code>sampleRate</code> attribute of the node's
+                <a>AudioContext</a>.
+                </li>
+                <li>Let <code>dt</code> be <code>1 / contextRate</code>.
+                </li>
+                <li>Let <code>index</code> be 0.
+                </li>
+                <li>Let <code>computedPlaybackRate</code> be <code>playbackRate
+                * pow(2, detune / 1200)</code> . This combines the two k-rate
+                parameters affecting playback rate into a single computed
+                quantity.
+                </li>
+                <li>While <code>index</code> is less than the length of the
+                audio block to be rendered:
+                  <ol>
+                    <li>If <code>currentTime &lt; start</code> or
+                    <code>currentTime &gt;= stop</code>, emit silence for the
+                    output frame at <code>index</code>.
+                    </li>
+                    <li>Else,
+                      <ol>
+                        <li>If <code>bufferTime</code> &lt; 0, set
+                        <code>bufferTime</code> to <code>offset + ((currentTime
+                        - start) * computedPlaybackRate)</code>. This is the
+                        <a>playhead position</a> for the first output sample
+                        frame.
+                        </li>
+                        <li>Let <code>effectiveBufferTime</code> equal
+                        <code>bufferTime</code>.
+                        </li>
+                        <li>map the playhead position to the correct point
+                        within the loop body accounting for loop iterations:
+                          <ol>
+                            <li>While <code>loop</code> is true and
+                            <code>effectiveBufferTime</code> &gt;=
+                            <code>actualLoopEnd</code>,
+                              <ol>
+                                <li>Let <code>effectiveBufferTime</code> equal
+                                <code>effectiveBufferTime - (actualLoopEnd -
```

Comment:
To further clarify and maybe avoid the next question, here's an assertion: if `effectiveBufferTime` is less than `actualLoopEnd`, then one of the following cases is true:

1. The playhead is within the body of the loop. No need to do anything, because the purpose of the loop is to map positions in _non-initial loop iterations_ into the equivalent positions within the loop body.

2. The playhead is prior to the body of the loop. No need to do anything, because the loop is not involved.

3. The playhead is prior to the buffer altogether (i.e. negative). No need to do anything, because silence will be output regardless.